workshop.md

```diff
@@ -91,162 +95,173 @@ Once you are in the new Lakehouse, create a new notebook by selecting `Open Note
+print(f"MODULE_NAME: {MODULE_NAME}")
+
+# ----------------- Download Functions -----------------
+
+# Function to download and save files
+def download_and_save_file(file_url, lakehouse_save_path):
+    try:
+        response = requests.get(file_url)
+        if response.status_code == 200:
+            # Create directories in Lakehouse if they don't exist
+            dir_path = os.path.dirname(lakehouse_save_path)
+            mssparkutils.fs.mkdirs(dir_path)
+            # Save the file content to Lakehouse
+            mssparkutils.fs.put(lakehouse_save_path, response.text, overwrite=True)
+            print(f"Successfully downloaded {lakehouse_save_path}")
+        else:
+            print(f"Failed to download {file_url}: HTTP {response.status_code}")
+    except Exception as e:
+        print(f"Error downloading {file_url}: {e}")
+
+# Function to recursively download contents
+def download_contents(contents, base_save_path):
+    for item in contents:
+        if item['type'] == 'file':
+            file_name = item['name']
+            download_url = item['download_url']
+            # Compute the relative path of the file within the module
+            relative_path = item['path'].replace(REPO_PATH + '/', '')
+            # Construct the Lakehouse save path
+            lakehouse_save_path = f"{base_save_path}/{relative_path}"
+            # Download and save the file
+            download_and_save_file(download_url, lakehouse_save_path)
+        elif item['type'] == 'dir':
+            # Handle subdirectories recursively
+            sub_dir_path = item['path']
+            sub_api_url = item['url']
+            sub_response = requests.get(sub_api_url)
+            if sub_response.status_code == 200:
+                sub_contents = sub_response.json()
+                # Compute the new base save path for the subdirectory
+                sub_base_save_path = f"{base_save_path}/{item['name']}"
+                download_contents(sub_contents, sub_base_save_path)
+            else:
+                print(f"Failed to list contents of {sub_dir_path}: HTTP {sub_response.status_code}")
+
+# Main execution
+def main():
+    # Construct the Lakehouse base save path for the module
+    lakehouse_module_path = f"{MARKDOWN_PATH}/{REPO_PATH}"
+    # Create the base directory in Lakehouse
+    mssparkutils.fs.mkdirs(lakehouse_module_path)
+    
+    # Get the contents of the folder via GitHub API
+    api_url = f"{GITHUB_API_URL}/{OWNER}/{REPO}/contents/{REPO_PATH}?ref={BRANCH}"
+    response = requests.get(api_url)
+    if response.status_code == 200:
+        contents = response.json()
+        download_contents(contents, lakehouse_module_path)
+    else:
+        print(f"Failed to list contents of {REPO_PATH}: HTTP {response.status_code}")
+        print(f"Response: {response.text}")
+
+# Run the main function
+main()
+
+# ----------------- List Markdown Files -----------------
+
+# 2. List markdown files in the includes folder
```

Comment:
Here for instance you have a great candidate for a new cell. It is already set apart by a comment, so why not make it its own cell, with text above explaining what the user is doing. "Now we'll use the `mssparkutils` library we included above to list the Markdown files we just downloaded", etc